test/nn/test_convolution.py

```diff
@@ -1677,6 +1679,7 @@ def test_conv_double_backward_stride(self):
                 )
 
     @dtypes(torch.float, torch.cfloat)
+    @dtypesIfMPS(torch.float)  # Complex not supported on MacOS13
```

Comment:
How about something like that?
```suggestion
    @dtypesIfMPS(*([torch.float] if product_version < 14.0 else [torch.float, torch.cfloat]))  # Complex not supported on MacOS13
```

Reply:
Ah thanks, I wasn't sure if conditioning datatypes on OS version was ok or not. Adding that now